### PR TITLE
New lexical attribute: is_currency

### PR DESCRIPTION
**Pull request by DuyguA**
As already discussed in  #1439 , I'd like to add is_currency lexical attribute under lang/. 

## Description
I made use of unicodedata.categories, currency symbols have category 'Sc'. [This category](https://www.fileformat.info/info/unicode/category/Sc/list.htm) has 54 currency symbols.

I added the test under tests/lang/test_attrs.py

Note that **is_currency** returns True only for currency symbols. Currency words e.g. dollar, euro, cent; I'll override in lang subhooks.  

I can add an update to [Lexeme API](url) as well. 

### Types of change
New feature.

## Checklist

- [x ] I have submitted the spaCy Contributor Agreement.
- [x ] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: spacy/attrs.pxd [hunk 1]
```diff
@@ -18,6 +18,7 @@ cdef enum attr_id_t:
     IS_QUOTE
     IS_LEFT_PUNCT
     IS_RIGHT_PUNCT
+    IS_CURRENCY
 
     FLAG18 = 18
```
Comment (ines) [inline spacy/attrs.pxd]: In order to not confuse the order of flags, we should remove this line and replace the next one with:

```python
FLAG19 = 19
```

File: spacy/attrs.pyx [hunk 2]
```diff
@@ -21,6 +21,7 @@ IDS = {
     "IS_QUOTE": IS_QUOTE,
     "IS_LEFT_PUNCT": IS_LEFT_PUNCT,
     "IS_RIGHT_PUNCT": IS_RIGHT_PUNCT,
+    "IS_CURRENCY": IS_CURRENCY,
     "FLAG18": FLAG18,
```
Comment (ines) [inline spacy/attrs.pyx]: Same here - essentially, the new `IS_CURRENCY` flag replaces `FLAG18`.

File: spacy/attrs.pxd (hunk 1 above)
Comment (ines) [inline spacy/attrs.pxd]: In order to not confuse the order of flags, we should remove this line and replace the next one with:

```python
FLAG19 = 19
```

**Comment by DuyguA** [comment]
Ooooopsi, flags messed up 😱
Kapish, I'll make the changes soon and notify from here, I hope that works.

**Pull request closed without merging**